src/Microsoft.AspNetCore.Mvc.Core/ContentResult.cs

```diff
@@ -67,13 +68,30 @@ public override Task ExecuteResultAsync(ActionContext context)
 
             if (Content != null)
             {
-                var bufferingFeature = response.HttpContext.Features.Get<IHttpBufferingFeature>();
-                bufferingFeature?.DisableResponseBuffering();
+                response.ContentLength = resolvedContentTypeEncoding.GetByteCount(Content);
 
-                return response.WriteAsync(Content, resolvedContentTypeEncoding);
-            }
+                var byteBuffer = ArrayPool<byte>.Shared.Rent(DefaultByteBufferSize);
```

Comment:
I would suggest that you put the body of this action result into an action result executor facade (see ViewResult, FileResult, etc.) and require the array pool instance on the constructor of the executor rather than relying on the static instance. That's the pattern we've followed elsewhere.

/cc @rynowak